common/common.hpp

```diff
@@ -42,6 +44,8 @@ inline static string StringFromByteVector(const vector<uint8_t> &vec) {
 	return string(vec.begin(), vec.end());
 }
 
+mender::common::expected::ExpectedLong StringToLong(string str, int base = 10);
```

Comment:
`const` is not really useful for copy-by-value arguments, other than protecting me from modifying a local copy. It's uncommon to use const in this context, so I think my preference would be not to.

You did however remind me that the string should be `const &`.

Reply:
haha, good!

But it does read out loud, that this value will not be modified right? So, bit of an odd default

Reply:
I *think* Rust has it opposite, doesn't it? You need to use `mut` to change anything after creation.

The style guide was frightfully indecisive in what is preferred 🙂:

> Using const on local variables is neither encouraged nor discouraged.

But since it doesn't mention anything, I checked what the standard library does, and [they use non-const everywhere](https://en.cppreference.com/w/cpp/container/vector/resize), so I think I'll stick to that. Especially since it's often useful to change it.

Reply:
Yeh, rust defaults to conts.

And makes sense, let's not challenge N years of experience doing this :smile: